packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -121,8 +122,11 @@ class DataSourceWithBackend<
   TQuery extends DataQuery = DataQuery,
   TOptions extends DataSourceJsonData = DataSourceJsonData,
 > extends DataSourceApi<TQuery, TOptions> {
+  protected userStorage: UserStorage;
+
   constructor(instanceSettings: DataSourceInstanceSettings<TOptions>) {
     super(instanceSettings);
+    this.userStorage = new UserStorage(instanceSettings.type);
```

Comment:
Can we always rely on that there is a `instanceSettings.type` on this line?

Reply:
yes, this value comes from the database, in which it's a mandatory field:
https://github.com/grafana/grafana/blob/exposeUserStorage/pkg/services/datasources/models.go#L154

Reply:
Awesome, thanks 👍